### PR TITLE
chore: Split up docker files and upgrade docker compose

### PR DESCRIPTION
**Pull request by ghost**
This PR:
- upgrades pizza and workflow OS to Ubuntu 22
- updates docker compose to use the plugin instead of the standalone (`docker compose` instead of `docker-compose`)
- creates a separate docker-compose.yml for development (no more restarting everything to pick up an api change. **Note**: this means that all devs will need to have this plugin and be able to run `docker compose`).
- adds a root set of pnpm scripts for starting and stopping containers
- prefer bash instead of shell for all scripts

### CODE REVIEW

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]
Apologies for taking so long to get back to this one.

One very minor request to change, but this works great locally and on the Pizza - thank you for picking this up!

File: package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "scripts": {
     "docker:up": "docker compose -f ./docker-compose.yml -f ./docker-compose.local.yml up -d",
-    "docker:down": "docker compose -f ./docker-compose.yml -f ./docker-compose.local.yml down --remove-orphans -v"
+    "docker:down": "docker compose -f ./docker-compose.yml -f ./docker-compose.local.yml down",
+    "docker:down-hard": "docker compose -f ./docker-compose.yml -f ./docker-compose.local.yml down --remove-orphans -v"
```
Comment (DafyddLlyr) [inline package.json:5]: Perfect - thank you 👍

File: README.md [hunk 2]
```diff
@@ -26,31 +27,28 @@ planx-new is a monorepo containing our full application stack. Here's a quick su
 
 1. Clone this repository. You'll need to get some additional environment variable files that aren't published here. Find instructions for copying these in 1password under the AWS Staging IAM user role entry
 
-1. Run the following command to get everything (postgres, sharedb, api and hasura server processes) up and running `docker-compose up --build -d`
-
-1. Move into the hasura directory `cd ../hasura.planx.uk` & install dependencies `pnpm i`. 
+1. Run `pnpm docker:up` from the project root to get everything (postgres, sharedb, api and hasura server processes) up and running and `pnpm docker:down` to stop all services. `pnpm docker:down-hard` will remove volumes (i.e. database data) and can be a useful hard reset when necessary.
 
-1. Open [Hasura's](https://hasura.io/) web console `pnpm start` & check that your Google email address is in the `users` table, if not then add it
+1. Open [Hasura's](https://hasura.io/) web console `http://localhost:7000` & check that your Google email address is in the `users` table, if not then add it
```
Comment (jessicamcinchak) [inline README.md:32]: should this be localhost:9695 for Hasura? or if it's changed to 7000 now, could env.example `HASURA_CONSOLE_PORT=9695` be updated please?

**Review by jessicamcinchak — APPROVED** [review]
Thanks for these changes & upgrades! 

I previously only had Docker Compose standalone but was able to install the Plugin no issues and run these changes locally :+1:

**Pull request merged**